apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs

```diff
@@ -29,50 +30,27 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
   @account_locked_response {:error, %{"code" => -32_000, "message" => "authentication needed: password or unlock"}}
 
   setup do
-    {:ok, empty} =
-      Core.new(
-        mined_child_block_num: 0,
-        known_hashes: [],
-        top_mined_hash: <<0::256>>,
-        parent_height: 0,
+    {:ok, empty} = Core.new(0, [], <<0::256>>, 0, child_block_interval: @child_block_interval)
+
+    {:ok, empty_high_max_gas_price} =
+      Core.new(0, [], <<0::size(256)>>, 0,
         child_block_interval: @child_block_interval,
-        chain_start_parent_height: 1,
-        block_submit_every_nth: 1,
-        finality_threshold: 12,
-        last_enqueued_block_at_height: 0
+        gas_price_adj_params: %GasPriceAdjustment{max_gas_price: 40_000_000_000}
       )
 
-    empty_with_gas_params = %{empty | formed_child_block_num: 5 * @child_block_interval, gas_price_to_use: 100}
+    empty_with_gas_params = %{empty | formed_child_block_num: 5000, gas_price_to_use: 100}
 
-    {:do_form_block, empty_with_gas_params} =
-      Core.set_ethereum_status(empty_with_gas_params, 1, 3 * @child_block_interval, false)
+    {:do_form_block, empty_with_gas_params} = Core.set_ethereum_status(empty_with_gas_params, 1, 3000, false)
 
     # assertions - to be explicit how state looks like
-    child_block_mined = 3 * @child_block_interval
+    child_block_mined = 3000
     assert {1, ^child_block_mined} = empty_with_gas_params.gas_price_adj_params.last_block_mined
 
-    {:ok, %{empty: empty, empty_with_gas_params: empty_with_gas_params}}
-  end
-
-  # Create the block_queue new state with non-initial parameters like it was recovered from db after restart / crash
-  # If top_mined_hash parameter is ommited it will be generated from mined_child_block_num
-  defp recover(known_hashes, mined_child_block_num, top_mined_hash \\ nil) do
-    top_mined_hash = top_mined_hash || "#{Kernel.inspect(trunc(mined_child_block_num / 1000))}"
-
-    Core.new(
-      mined_child_block_num: mined_child_block_num,
-      known_hashes: known_hashes,
-      top_mined_hash: top_mined_hash,
-      parent_height: 10,
-      child_block_interval: 1000,
-      chain_start_parent_height: 1,
-      block_submit_every_nth: 1,
-      finality_threshold: 12,
-      last_enqueued_block_at_height: 0
-    )
+    {:ok,
+     %{empty: empty, empty_with_gas_params: empty_with_gas_params, empty_high_max_gas_price: empty_high_max_gas_price}}
   end
 
-  describe "Block queue." do
+  describe "child_block_nums_to_init_with/4" do
```

Comment:
✨

Reply:
now we need an explanation what the function name means :D